src/OutputWriters/checkpointer.jl

```diff
@@ -217,7 +217,7 @@ function set!(model, filepath::AbstractString)
             if string(name) ∈ keys(file) # Test if variable exist in checkpoint.
                 model_field = model_fields[name]
                 halo = halo_size(model_field.grid)
-                loc  = location(model_field)
+                loc  = map(instantiate, location(model_field))
```

Comment:
is this what `JLD2OutputWriter` does?

Reply:
this a major change... might be the right one but probably should writing something in the comments about why

Reply:
Yeah I was just testing something, I have problem restarting with different halos. I have removed that behavior and hard coded the different halo loading in my repo manually, but I ll open an issue so we can discuss it, maybe it's something we want to support